.github/workflows/ci.yml

```diff
@@ -61,6 +68,111 @@ jobs:
           pre-commit run --all-files
           pytest taf/tests
 
+      - name: Set Benchmark Threshold from File
+        run: |
+          CURRENT_BRANCH=${GITHUB_REF#refs/heads/}
+
+          THRESHOLD=$(jq --arg branch "$CURRENT_BRANCH" '.[$branch]' ./benchmark_thresholds.json)
+
+          if [ "$THRESHOLD" == "null" ]; then
+            THRESHOLD=10  # Default threshold if not specified
+          fi
+          echo "BENCHMARK_THRESHOLD=$THRESHOLD" >> $GITHUB_ENV
+          echo "Using benchmark threshold: $THRESHOLD"
+
+      - name: Display Benchmark Threshold
+        run: echo "Benchmark Threshold is $BENCHMARK_THRESHOLD"
+
+      - name: Fetch the latest artifact by name and branch
+        id: get_artifact
+        uses: actions/github-script@60a0d83039c74a4aee543508d2ffcb1c3799cdea #@v7.01
+        with:
+          script: |
+            const artifacts = await github.rest.actions.listArtifactsForRepo({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+            });
+            const filteredArtifacts = artifacts.data.artifacts.filter(artifact => artifact.name === process.env.ARTIFACT_NAME);
+            let latestArtifact = null;
+
+            for (const artifact of filteredArtifacts) {
+              const run = await github.rest.actions.getWorkflowRun({
+                owner: context.repo.owner,
+                repo: context.repo.repo,
+                run_id: artifact.workflow_run.id,
+              });
+
+              if (run.data.head_branch === process.env.BRANCH_NAME) {
+                if (!latestArtifact || new Date(artifact.created_at) > new Date(latestArtifact.created_at)) {
+                  latestArtifact = artifact;
+                }
+              }
+            }
+
+            if (latestArtifact) {
+              console.log(`Found latest artifact: ${latestArtifact.id}`);
+              core.setOutput('artifact_id', latestArtifact.id.toString());
+            } else {
+              console.log('No matching artifacts found.');
+              core.setOutput('artifact_id', '');
+            }
```

Comment:
question: could we check in this javascript (and other) code somewhere in the `.github` (e.g. `.github/scripts`) directory and have GitHub actions require the file? The benefit is that we have syntax highlighting and would be easier to debug [1]?

[1] - https://github.com/orgs/community/discussions/26248